software/firmware/configuration.py

```diff
@@ -192,20 +192,16 @@ def load_config(cls, config_spec: ConfigSpec):
         """If this class has config points, this method validates and returns the config dictionary
         as saved in this class's config file, else, returns an empty dict."""
         if len(config_spec):
-            data = load_file(ConfigFile.config_filename(cls))
-            config = config_spec.default_config()
-            if not data:
-                return config
-            else:
-                saved_config = load_json_data(data)
-                validation = config_spec.validate(saved_config)
+            saved_config = load_json_file(ConfigFile.config_filename(cls))
+            default_config = config_spec.default_config()
+            validation = config_spec.validate(saved_config)
 
-                if not validation.is_valid:
-                    raise ValueError(validation.message)
+            if not validation.is_valid:
+                raise ValueError(validation.message)
 
-                config.update(saved_config)
+            default_config.update(saved_config)
```

Comment:
[optional] I feel like this variable name is misleading now (as well as the return statement below. Once this update happens it is no longer the default config. Perhaps it should just be `config`.

Reply:
Change made.